azext_iot/operations/hub.py

```diff
@@ -398,11 +398,9 @@ def _parse_auth(parameters):
 
 
 def iot_device_module_list(cmd, device_id, hub_name=None, top=1000, resource_group_name=None, login=None):
-    query = "select * from devices.modules where devices.deviceId = '{}'".format(device_id)
-    result = iot_query(cmd, query, hub_name, top, resource_group_name, login=login)
-    if not result:
-        logger.info('No modules found on registered device "%s".', device_id)
-    return result
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    service_sdk, errors = _bind_sdk(target, SdkType.service_sdk)
+    return service_sdk.get_modules_on_device(device_id)[:top]
```

Comment:
Let's add some [error handling](https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/operations/hub.py#L444-L447) here.

Reply:
This should also help with the linter errors in the CI build

Reply:
Implemented exception handling